scipy/special/cephes/hyp2f1.c

```diff
@@ -590,3 +595,27 @@ static double hyp2f1ra(double a, double b, double c, double x,
 
     return f0;
 }
+
+
+/*
+    15.4.2 Abramowitz & Stegun.
+*/
+static double hyp2f1_neg_c_equal_bc(double a, double b, double x)
+{
+    double k;
+    double err;
+    double collector = 1;
+    double sum = 1;
+    double const max_err = (1e-7/1e-16) - 1;
+
+    for (k = 1; k <= -b; k++) {
```

Comment:
This needs a check along the lines of 
`if (!(fabs(b) < 1e15)) { return NPY_NAN; }`, otherwise it enters an infinite loop when you run out bits in the mantissa.